cfe_internal/enterprise/federation/federation.cf

```diff
@@ -0,0 +1,210 @@
+        perms => mo("600", "root");
+      "$(federation_config.federation_dir)/fedhub/dump/."
+        create => "true",
+        perms => mo("600", "root");
+      "$(federation_config.federation_dir)/fedhub/transport/."
+        create => "true",
+        perms => mo("600", "root");
+
+    # TODO: Render these from templates:
+    federation_am_feeder::
+      "$(federation_config.bin_dir)/dump.sh"
+        create => "true",
+        perms => mo("600", "root");
+    federation_am_transporter::
+      "$(federation_config.bin_dir)/transport.sh"
+        create => "true",
+        perms => mo("600", "root");
+    federation_am_superhub::
+      "$(federation_config.bin_dir)/import.sh"
+        create => "true",
+        perms => mo("600", "root");
+}
+
+bundle agent federation_manage_database
+{
+  files:
+    "/var/cfengine/state/pg/data/postgresql.conf"
+      edit_line => insert_lines("shared_buffers=1GB");
+    "/var/cfengine/state/pg/data/postgresql.conf"
+      edit_line => insert_lines("max_locks_per_transaction=4000");
+}
+
+bundle agent federation_manage_dump
+{
+  commands:
+    federation_am_feeder::
+      # TODO: pass correct arguments once script is done
+      "bash $(federation_config.bin_dir)/dump.sh"
+        contain => in_shell;
+}
+
+bundle agent federation_manage_transport
+{
+  commands:
+    federation_am_transporter::
+      # TODO: pass correct arguments once script is done
+      "bash $(federation_config.bin_dir)/transport.sh"
+        contain => in_shell;
+}
+
+bundle agent federation_manage_import
+{
+  commands:
+    federation_am_superhub::
+      # TODO: pass correct arguments once script is done
+      "bash $(federation_config.bin_dir)/import.sh"
+        contain => in_shell;
+}
+
+bundle agent federation_entry_bundle
+{
+  meta:
+    (policy_server|am_policy_hub).enterprise_edition::
+      "tags"
+        slist => { "enterprise_maintenance" };
+  classes:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_config_exists"
+        expression => fileexists("/opt/cfengine/federation/cfapache/federation-config.json");
+  methods:
+    federation_config_exists::
+      "federation_config"
+        usebundle => federation_config;
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_manage_files"
+        usebundle => federation_manage_files;
+    federation_am_superhub::
+      "federation_manage_cfdrop"
+        usebundle => federation_manage_cfdrop;
+    federation_enabled::
+      "federation_manage_database"
+        usebundle => federation_manage_database;
+      "federation_manage_dump"
+        usebundle => federation_manage_dump;
+      "federation_manage_transport"
+        usebundle => federation_manage_transport;
+      "federation_manage_import"
+        usebundle => federation_manage_import;
+  reports:
+    !enterprise_edition::
+      "Federated reporting is only available in CFEngine Enterprise.";
+    !(policy_server|am_policy_hub)::
```

Comment:
Maybe only say federated reporting available on policy server/hub when enterprise_edition is defined.